onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -7738,29 +7738,59 @@ def aten_scalar_tensor_sym_number(
 
 @torch_op("aten::scatter.src", trace_only=True)
 def aten_scatter_src(
-    self: TReal,
+    self: TensorType,
     dim: int,  # we have to use int here because ScatterElements() will use this attribute
     index: TInt,
-    src: TReal,
-) -> TReal:
+    src: TensorType,
+) -> TensorType:
     """scatter.src(Tensor self, int dim, Tensor index, Tensor src) -> Tensor"""
     return op.ScatterElements(self, index, src, axis=dim)
 
 
-@torch_op("aten::scatter.value", trace_only=True)
-def aten_scatter_value(
-    self: TReal,
+@torch_op("aten::scatter.value", trace_only=True, private=True)
+def aten_scatter_value_impl(
+    self: TensorType,
     dim: int,  # we have to use int here because ScatterElements() will use this attribute
     index: TInt,
-    value: TReal,
-) -> TReal:
+    value: Any,
```

Comment:
```suggestion
    value: float,
```

This is ok and will handle bools etc.